lib/rdf/model/uri.rb

```diff
@@ -717,7 +717,7 @@ def end_with?(string)
     # @param  [RDF::URI] other
     # @return [Boolean] `true` or `false`
     def eql?(other)
-      other.is_a?(URI) && self.hash == other.hash && self == other
+      other.is_a?(URI) && self == other
```

Comment:
It was changed to first check hashes then values as a hash comparison is faster (once it's been computed), so the existing behavior is what was intended.